src/main/java/perosnal/spotifymatcher/service/UserService.java

```diff
@@ -0,0 +1,86 @@
+package perosnal.spotifymatcher.service;
+
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import lombok.RequiredArgsConstructor;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import perosnal.spotifymatcher.model.User;
+import perosnal.spotifymatcher.api.SpotifyAPI;
+import perosnal.spotifymatcher.repository.UserRepository;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+
+    public List<User> match(String accessToken) {
+        User user = userRepository.findByTokenAccessToken(accessToken);
+        List<User> matches = filterMatches(user, userRepository.getMatches(user.getId(), 3));
+        if (user.getBiography()==null) {
+            return null;
+        }
+        if (matches.size() > 0) {
+            user.getMatches().addAll(matches.stream().map(User::getId).collect(Collectors.toList()));
```

Comment:
[IDE] Please pay more attention to [your IDE warnings](https://www.jetbrains.com/help/idea/code-inspection.html). This code shows:

> 'collect(toList())' can be replaced with 'toList()' 

ᴛʜɪs ᴀᴄᴛɪᴏɴ ᴡᴀs ᴘᴀʀᴛɪᴀʟʟʏ ᴀᴜᴛᴏᴍᴀᴛᴇᴅ ʙʏ [GɪᴛHᴜʙ's ʀᴇsᴘᴏɴsᴇ ᴛᴇᴍᴘʟᴀᴛᴇs](https://github.com/settings/replies)